src/core/thread/mesh_forwarder.cpp

```diff
@@ -63,6 +63,17 @@ void ThreadLinkInfo::SetFrom(const Mac::RxFrame &aFrame)
         IgnoreError(aFrame.GetDstPanId(mPanId));
     }
 
+    {
+        Mac::PanId dstPanId;
+
+        if (kErrorNone != aFrame.GetDstPanId(dstPanId))
+        {
+            dstPanId = mPanId;
+        }
+
+        mIsPanIdBroadcast = (dstPanId == Mac::kPanIdBroadcast);
```

Comment:
`dstPanId` variable can used without any value assigned to it if the `GetDstPanId()` returns error.

Move `mIsPanIdBroadcast` to inside `if` block and set `mIsPanIdBroadcast` to false if we fail to get a dst pan id?

Reply:
The assumption is if the destination PAN ID is not present, it is identical to the source PAN ID. mPanId is cleared at the beginning of this method. If both PAN IDs are missing, dstPanId will be zero the same as the mPanId.

Reply:
I see. It make sense. Thanks. The code is good as is.